doc/release/upcoming_changes/18880.compatibility.rst

```diff
@@ -0,0 +1,34 @@
+Ufunc ``signature=...`` and ``dtype=`` generalization and ``casting``
+---------------------------------------------------------------------
+The behaviour for ``signature=...`` and ``dtype=...`` can differ in
+some cases to the previous behaviour.
```

Comment:
```suggestion
The behaviour for ``np.ufunc(1.0, 1.0, signature=...)`` and ``np.ufunc(1.0, 1.0, dtype=...)``
can yield different loops in 1.21 than in 1.20 because of changes in promotion. 
```

Reply:
Adopted as:
> The behaviour for ``np.ufunc(1.0, 1.0, signature=...)`` or
``np.ufunc(1.0, 1.0, dtype=...)`` can now yield different loops in 1.21 compared
to 1.20 because of changes in promotion.